symengine/mp_class.h

```diff
@@ -4,14 +4,18 @@
 #include <symengine/symengine_config.h>
 #include <symengine/mp_wrapper.h>
 
-#if SYMENGINE_INTEGER_CLASS == SYMENGINE_PIRANHA
+#ifdef HAVE_SYMENGINE_PIRANHA
 #include <piranha/mp_integer.hpp>
 #include <piranha/mp_rational.hpp>
```

Comment:
Yes, probably. But these are needed for `to_integer_class` conversions to work. Consider the case when `integer_class` is `fmpz_wrapper` and the user is using `UIntPolyPiranha`. We need a mechanism to convert from `piranha::integer` to `fmpz_wrapper`, and thus piranha functions are required.

Same argument for the `#if SYMENGINE_INTEGER_CLASS == SYMENGINE_FLINT` -> `#ifdef HAVE_SYMENGINE_FLINT` change.